torch/distributions/binomial.py

```diff
@@ -0,0 +1,94 @@
+from numbers import Number
+import torch
+import math
+from torch.distributions import constraints
+from torch.distributions.distribution import Distribution
+from torch.distributions.utils import broadcast_all, probs_to_logits, lazy_property
+from torch.distributions.utils import clamp_probs
+from torch.autograd import Variable
+
+
+class Binomial(Distribution):
+    r"""
+    Creates a Binomial distribution parameterized by `total_count` and
+    either `probs` or `logits` (but not both).
+
+    -   :meth:`sample` requires a single shared `total_count` for all
+        parameters and samples.
+
+    Example::
+
+        >>> m = Binomial(100, torch.Tensor([0 , .2, .8, 1]))
+        >>> x = m.sample()
+         21
+         24
+         30
+         25
+        [torch.FloatTensor of size 4]]
+
+    Args:
+        total_count (int): number of trials
+        probs (Tensor or Variable): event probabilities
+        logits (Tensor or Variable): event log probabilities
+    """
+    params = {'probs': constraints.unit_interval}
+    has_enumerate_support = True
+
+    def __init__(self, total_count=1, probs=None, logits=None):
+        if not isinstance(total_count, Number):
+            raise NotImplementedError('inhomogeneous total_count is not supported')
+        self.total_count = total_count
+        if (probs is None) == (logits is None):
+            raise ValueError("Either `probs` or `logits` must be specified, but not both.")
+        if probs is not None:
+            self.probs, = broadcast_all(probs)
+        else:
+            self.logits, = broadcast_all(logits)
+
+        probs_or_logits = probs if probs is not None else logits
+        if isinstance(probs_or_logits, Number):
+            batch_shape = torch.Size()
+        else:
+            batch_shape = probs_or_logits.size()
+        super(Binomial, self).__init__(batch_shape)
+
+    @constraints.dependent_property
+    def support(self):
+        return constraints.integer_interval(0, self.total_count)
+
+    @lazy_property
+    def logits(self):
+        return probs_to_logits(self.probs)
+
+    @lazy_property
+    def probs(self):
+        return self.logits.exp()
+
+    def sample(self, sample_shape=torch.Size()):
+        shape = (self.total_count,) + self._extended_shape(sample_shape)
+        return torch.bernoulli(self.probs.expand(shape)).sum(dim=0)
+
+    def log_prob(self, value):
+        self._validate_log_prob_arg(value)
+        probs = clamp_probs(self.probs)
+        log_factorial_n = math.lgamma(self.total_count + 1)
+        log_factorial_k = torch.lgamma(value + 1)
+        log_factorial_nmk = torch.lgamma(self.total_count - value + 1)
+        return (log_factorial_n - log_factorial_k - log_factorial_nmk +
+                value * self.logits + (self.total_count - value) * torch.log1p(-probs))
+
+    def entropy(self):

```

Comment:
This is only approximate, right? I feel like we'd be better off leaving it not implemented than implementing an approximate entropy.

Also, we should adapt the TestKL tests to entropy tests to catch things like this.